tests/python_package_test/test_dask.py

```diff
@@ -656,6 +658,166 @@ def test_ranker(output, client, group):
     client.close(timeout=CLIENT_CLOSE_TIMEOUT)
 
 
+@pytest.mark.parametrize('task', tasks)
+@pytest.mark.parametrize('output', ['array', 'dataframe'])
```

Comment:
why are you omitting sparse matrices? Did you run into some issues with them?

Reply:
No reason other than to keep number of tests reasonable, exponential grid size with pytest.mark

Reply:
right, right gotchu. Ok for now I'd rather err on the side of more tests since this change is fairly large and touching parts of distributed LightGBM training that haven't been tested. Could you add all the data output types back? Let's see how much it changes the total runtime.

Reply:
Addressed in b11db84